queries.yaml

```diff
@@ -822,22 +845,58 @@ post_process:
       # table.
       min_zoom: >-
         [min_zoom for min_zoom, area_threshold in [
-            (4,  40000000000),
-            (5,  10000000000),
-            (6,   5000000000),
-            (7,    400000000),
-            (8,    200000000),
-            (9,    100000000),
-            (10,    10000000),
-            (11,     4000000),
-            (12,      750000),
-            (13,      100000),
-            (14,       50000),
-            (15,       10000),
-            (16,        None),
+            (1, 1000000000000),
+            (2,  500000000000),
+            (3,  250000000000),
+            (4,  120000000000),
+            (5,   80000000000),
+            (6,   40000000000),
+            (7,   10000000000),
+            (8,     500000000),
+            (9,     200000000),
+            (10,     40000000),
+            (11,      8000000),
+            (12,      1000000),
+            (13,       500000),
+            (14,        50000),
+            (15,        10000),
+            (16,         None),
         ] if area >= area_threshold][0]
       where: >-
-        kind == 'lake' and label_placement
+        kind in ['lake','bay','water','riverbank','reservoir'] and label_placement
+
+  # now that we have the label points
+  # drop most water properties at lower zooms
+  # this is for lines and polygons
+  - fn: vectordatasource.transform.drop_properties
+    params:
+      source_layer: water
+      start_zoom: 8
+      end_zoom: 15
+      properties:
+        - id
+        - area
+        - layer
+        - wikidata_id
+        - kind_tile_rank
+        - label_placement
+      where: >-
```

Comment:
Also drop `boundary`

Reply:
This where clause doesn't seem to work.

There's another pattern we can switch to, like:

```
  # no lake labels at zoom 0-3:
  # https://github.com/tilezen/vector-datasource/issues/1730
  - fn: vectordatasource.transform.drop_names
    params:
      source_layer: water
      start_zoom: 0
      end_zoom: 4
      geom_types: [Polygon, MultiPolygon]
      where: >-
        kind == 'lake'
```

Reply:
Addressed via https://github.com/tilezen/vector-datasource/pull/2010/commits/77e663c183b3ff687215386ce9ad219a8301f4fd.